packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -32,43 +32,60 @@ export type AsyncTask = Task & {
   _zone: Zone;
 };
 
+// Delay of 50 ms to reset currentEventTracingZone.
+const RESET_TRACING_ZONE_DELAY = 50;
+
 export class InteractionTracker {
-  constructor() {
+  // Allows to track several events triggered by the same user interaction in the right Zone.
+  private currentEventTracingZone?: Zone = undefined;
 
+  constructor() {
     const runTask = Zone.prototype.runTask;
-    Zone.prototype.runTask = function(
+    Zone.prototype.runTask = (
       task: AsyncTask,
       applyThis: unknown,
       applyArgs: unknown
-    ) {
+    ) => {
       const time = Date.now();
 
       console.warn('Running task');
+      console.log(task);
       console.log(task.zone);
 
-      let taskZone = this;
+      let taskZone: Zone = Zone.current;
```

Comment:
The `: Zone` is implied by the assignment. What would you think about removing it?